shinkai-libs/shinkai-message-primitives/src/shinkai_message/shinkai_message_schemas.rs

```diff
@@ -865,3 +866,109 @@ impl fmt::Display for RegistrationCodeType {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::json;
+
+    #[test]
+    fn test_job_message_serialization() {
+        let job_message = JobMessage {
+            job_id: "test_job".to_string(),
+            content: "test content".to_string(),
+            parent: Some("parent_id".to_string()),
+            sheet_job_data: Some("sheet_data".to_string()),
+            tools: Some(vec!["tool1".to_string(), "tool2".to_string()]),
+            callback: Some(Box::new(CallbackAction::Job(JobMessage {
+                job_id: "callback_job".to_string(),
+                content: "callback content".to_string(),
+                parent: None,
+                sheet_job_data: None,
+                tools: None,
+                callback: None,
+                metadata: None,
+                tool_key: None,
+                fs_files_paths: vec![],
+                job_filenames: vec![],
+            }))),
+            metadata: Some(MessageMetadata {
+                tps: Some("10".to_string()),
+                duration_ms: Some("100".to_string()),
+                function_calls: Some(vec![FunctionCallMetadata {
+                    name: "test_function".to_string(),
+                    arguments: {
+                        let mut map = serde_json::Map::new();
+                        map.insert("arg1".to_string(), json!("value1"));
+                        map
+                    },
+                    tool_router_key: Some("router_key".to_string()),
+                    response: Some("function response".to_string()),
+                }]),
+            }),
+            tool_key: Some("specific_tool".to_string()),
+            fs_files_paths: vec![],
+            job_filenames: vec!["file1.txt".to_string()],
+        };
+
+        // Test serialization
+        let serialized = serde_json::to_string(&job_message).expect("Failed to serialize JobMessage");
+
+        // Test deserialization
+        let deserialized: JobMessage = serde_json::from_str(&serialized).expect("Failed to deserialize JobMessage");
+
+        assert_eq!(job_message, deserialized);
+    }
+
+    #[test]
+    fn test_job_message_minimal() {
```

Comment:
the test that i think it would helpful is one that converts from json to JobMessage so we make sure that we have backwards compatibility